.github/workflows/ci.yml

```diff
@@ -194,6 +194,43 @@ jobs:
           mode: walltime
           token: ${{ secrets.CODSPEED_TOKEN }}
 
+
+  compat-integration-test-memory:
```

Comment:
We can have multi-dimensional matrices, as well as some dimensions precising two things

We could combine everything in one big matrix, one dimension being the library we are testing, the other one the mode and the runner it runs on like here https://github.com/CodSpeedHQ/codspeed-cpp/blob/f23b68b27f8543f88f95ff0bdc46785733dee061/.github/workflows/ci.yml#L57-L62

WDYT ?

We could conditionnallly remove the .cargo/config.toml, as well as conditionnally skip the uploading part for memory (for now)